### PR TITLE
Use CustomScopeAnnotationsBuildItem where needed

### PR DESCRIPTION
**Pull request by mkouba**
- follows up on #7076

### CODE REVIEW

File: extensions/scheduler/deployment/src/main/java/io/quarkus/scheduler/deployment/SchedulerProcessor.java [hunk 1]
```diff
@@ -104,14 +104,14 @@ public boolean appliesTo(org.jboss.jandex.AnnotationTarget.Kind kind) {
...
+                        && (target.annotations().containsKey(SCHEDULED_NAME)
+                                || target.annotations().containsKey(SCHEDULES_NAME))) {
                     // Class with no scope annotation but with @Scheduled method
-                    if (context.getTarget().asClass().annotations().containsKey(SCHEDULED_NAME)
-                            || context.getTarget().asClass().annotations().containsKey(SCHEDULES_NAME)) {
-                        LOGGER.debugf("Found scheduled business methods on a class %s with no annotations - adding @Singleton",
-                                context.getTarget());
-                        context.transform().add(Singleton.class).done();
-                    }
+                    LOGGER.debugf("Found scheduled business methods on a class %s with no annotations - adding @Singleton",
```
Comment (jmartisk) [inline extensions/scheduler/deployment/src/main/java/io/quarkus/scheduler/deployment/SchedulerProcessor.java]: would be better to say "with no scope defined", there could be some other unrelated annotations

**Review by jmartisk — APPROVED** [review]
Experimented with it locally and got it working. Just left a minor note about a debug message.
A test would be nice too, but I'll give a +1 already :)

**Comment by mkouba** [comment]
> A test would be nice too..

@jmartisk I don't think we need a new test. It's just improving the existing code...

**Review by manovotn — APPROVED** [review]
Nice catch! Apparently in my original PR I missed a lot of cases which should have been updated.
You should have bullied me harder :-D

**Pull request merged**